src/main/modules/state-redirect/index.ts

```diff
@@ -37,4 +56,12 @@ export class StateRedirectMiddleware {
       })
     );
   }
+
+  private hasPartnerResponded(userCase: CaseWithId, isApplicant2: boolean) {
+    return !(
+      ((isApplicant2 && [State.AwaitingApplicant1Response, State.Applicant2Approved].includes(userCase?.state)) ||
+        (!isApplicant2 && userCase?.state === State.AwaitingApplicant2Response)) &&
+      dayjs(userCase.dueDate, 'MMMM Do YYYY').diff(dayjs()) < 0
+    );
```

Comment:
would it be easier to make this be hasPartnerNotResponded rather than negate it? 
 because if it returns True at the moment it doesn't necessarily mean they've responded, just that they haven't not responded

Reply:
i agree to the name change, the ! was done before i writed the logic/check and i negated it where the method is called [here](https://github.com/hmcts/nfdiv-frontend/pull/1159/files#diff-631965ac66b32e6255fecaff8f97fda20f61ce22335af920dd9f2bd642c9942aR31) but cleaning the name up is better.Thanks